### PR TITLE
stm32: add dma_get_number_of_data

### PR DESCRIPTION
**Pull request by DanielG**
Just something that's convenient when you need to know how much has been transferred so far.

### CODE REVIEW

**Comment by DanielG** [comment]
No response after three months. I guess I'm starting to see why that whole unicore-mx fork happened. Given how relatively small this patch is I'm really disappointed it's just getting ignored :/

**Comment by karlp** [comment]
It's not being ignored, but when the entire contribution consists of ```return DMA_CNDTR(dma,  channel)``` but you felt the need to go and add copyright claims all over the code it wasn't something I was super excited about. Is it a useful addition? sure! Is it a bug that needed immediate attention, or functionality that wasn't available without heavy work on the user side? not really.

**Comment by DanielG** [comment]
I was under the impression adding copyright headers is good form, especially since the GPL relies on copyright to implement it's protections. Naturally given the triviality of the patch I'm not against removing them.

Anyways I wasn't trying to attack anyone, I was just trying to point out that ignoring PRs outright looks really bad. I'm also a project maintainer, I know that sometimes you just don't have time to look at irrelevant things like this right away. The problem is I would actually like to contribute more than just this trivial patch in the future but if something this small gets ignored why should I bother working on anything more substantial?

**Comment by karlp** [comment]
I'm sorry I didn't give more immediate feedback.  I do always try.  But the copyright additions gave me a hard time.  I understand people like them, I know some projects (most apache) actively get _rid_ of them, I know that people get very heated on whether they're actually even required or not for any sort of enforcement.  My general approach is that it's noise.  The commit log tracks authorship, having a line for every edit isn't scaleable, and having a list of "owners" in a file discourages other people from editing.  Trying to find a nice way of saying, "I think the commit part looks fine, but I want you to throw out the attribution you wanted to add" was what stopped me at the time.  You also never came back to push on it, and yes, some requests do fall through the cracks.    I've put some more comments inline, and will try and get this into my queue, as it is indeed a small patch.

File: include/libopencm3/stm32/common/dma_common_l1f013.h [hunk 1]
```diff
@@ -422,4 +424,3 @@ END_DECLS
 /** @endcond */
 
 /**@}*/
-
```
Comment (karlp) [inline include/libopencm3/stm32/common/dma_common_l1f013.h]: unrelated whitespace changes are discouraged.

File: lib/stm32/common/dma_common_f24.c [hunk 2]
```diff
@@ -775,6 +776,20 @@ void dma_set_memory_address_1(uint32_t dma, uint8_t stream, uint32_t address)
 }
 
 /*---------------------------------------------------------------------------*/
+/** @brief DMA Stream Get the Transfer Block Size
+
+@param[in] dma unsigned int32. DMA controller base address: DMA1 or DMA2
+@param[in] stream unsigned int8. Stream number: @ref dma_st_number
+@returns unsigned int16. Number of remaining data words to transfer (65535
+maximum).
+*/
```
Comment (karlp) [inline lib/stm32/common/dma_common_f24.c]: There's no need for the [in] and the "unsigned int32" stuff.  Doxygen figures that out itself from the code.  Try and use the full * on each line like you should see in most other comments.

You can even drop the "@brief" and simply put a "." (full stop) at the end of the first line if you like.

File: lib/stm32/common/dma_common_f24.c [hunk 3]
```diff
@@ -791,4 +806,3 @@ void dma_set_number_of_data(uint32_t dma, uint8_t stream, uint16_t number)
 	DMA_SNDTR(dma, stream) = number;
 }
 /**@}*/
-
```
Comment (karlp) [inline lib/stm32/common/dma_common_f24.c]: whitespace again

File: lib/stm32/common/dma_common_f24.c (hunk 2 above)
Comment (karlp) [inline lib/stm32/common/dma_common_f24.c]: (I see that this existing code isn't using the full * comment style so props for at least keeping it consistent) :)

**Comment by karlp** [comment]
re: copyright headers, I can't find a better source right now, but as I remember it, the copyright headers were only useful for american source pre 1992, when the us finally adopted the berne convention.  http://geotoolsnews.blogspot.is/2016/07/copyright-headers-in-source-code-files.html has some, as does https://en.wikipedia.org/wiki/Copyright_notice but it's still a heated debate in many communities :)

**Comment by DanielG** [comment]
I think GNU basically strongly encourages putting the license and copyright notices in the source files but from reading the GPL it does sound like it's not required, though it does say:

> The work must carry prominent notices stating that you modified it, and giving a relevant date.

So whether a git commit is sufficient here is debatable. (See also https://www.gnu.org/licenses/gpl-faq.en.html#NoticeInSourceFile)

Of course you are right that authorship information already lives in git but the problem with that is end-user doesn't always get the whole git repo but rather just a tarball or something. 

I think at the very least an `AUTHORS` or `CONTRIBUTORS` file that just collects that info in a central place so it's not noise in the source could be useful.

File: lib/stm32/common/dma_common_f24.c (hunk 2 above)
Comment (DanielG) [inline lib/stm32/common/dma_common_f24.c]: I just based the description on the `_set_` call as closely as I could since I don't actually know the proper doxygen syntax ;)

So do you want me to fix the style of the rest of the stuff in the file or just keep it as is?

File: include/libopencm3/stm32/common/dma_common_l1f013.h (hunk 1 above)
Comment (DanielG) [inline include/libopencm3/stm32/common/dma_common_l1f013.h]: Sorry, my editor does that. I usually remove those changes again when submitting something but I though it doesn't really hurt at the end of the file since even a git blame wouldn't really show that anyways.

File: lib/stm32/common/dma_common_f24.c (hunk 2 above)
Comment (karlp) [inline lib/stm32/common/dma_common_f24.c]: don't worry about it, it's the same style as the rest of the file, even if the file is not matching the rest of the project.  At some point I'll go and clean them up.

File: include/libopencm3/stm32/common/dma_common_l1f013.h (hunk 1 above)
Comment (karlp) [inline include/libopencm3/stm32/common/dma_common_l1f013.h]: there's a "make stylecheck" target that is a bit noisy, but can be helpful.

**Comment by karlp** [comment]
thanks for this, it's now merged in https://github.com/libopencm3/libopencm3/commit/743513a4b1b15901fd6ff35f9b3c8812dc96dcaa
